packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -467,6 +441,37 @@ class ThemeData {
     );
   }
 
+  // See <https://www.w3.org/TR/WCAG20/#relativeluminancedef>
+  static double _linearizeColorComponent(double component) {
+    if (component <= 0.03928)
+      return component / 12.92;
+    return math.pow((component + 0.055) / 1.055, 2.4);
+  }
+
+  /// Determines whether the given [Color] is [Brightness.light] or
+  /// [Brightness.dark].
+  ///
+  /// This compares the luminosity of the given color to a threshold value that
+  /// matches the material design specification.
+  static Brightness estimateBrightnessForColor(Color color) {
+    // See <https://www.w3.org/TR/WCAG20/#relativeluminancedef>
+    final double R = _linearizeColorComponent(color.red / 0xFF);
```

Comment:
That line doesn't mask off the low 8 bits, it converts the integer representation (0 - 255) to a real interpretation (0.0 - 1.0).